examples/using_printer.rs

```diff
@@ -0,0 +1,84 @@
+use bat::{
```

Comment:
This example is quite long. Is there anything that it demonstrates that couldn't be done in the other example?

Reply:
The example demonstrate on how to print header, snip, and footer at certain condition.
This might be useful for custom error reporting like [codespan](https://github.com/brendanzab/codespan).